model/src/test/java/jetbrains/jetpad/model/composite/CompositesBetweenTest.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright 2012-2016 JetBrains s.r.o
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package jetbrains.jetpad.model.composite;
+
+import jetbrains.jetpad.test.BaseTestCase;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.List;
+
+import static java.util.Arrays.*;
+import static org.junit.Assert.*;
+
+public class CompositesBetweenTest extends BaseTestCase {
+  private SimpleComposite c, e, f, g, h, i, k, l, m, o, p, r, s, t, u, v, w, x, y;
+
+  @Before
+  public void init() {
+    new SimpleComposite("a",
+      new SimpleComposite("b",
+        e = new SimpleComposite("e"),
+        f = new SimpleComposite("f")),
+      c = new SimpleComposite("c",
+        g = new SimpleComposite("g")),
+      new SimpleComposite("d",
+        h = new SimpleComposite("h",
+          k = new SimpleComposite("k"),
+          l = new SimpleComposite("l"),
+          m = new SimpleComposite("m",
+            r = new SimpleComposite("r"),
+            s = new SimpleComposite("s"),
+            t = new SimpleComposite("t"))),
+        i = new SimpleComposite("i"),
+        new SimpleComposite("j",
+          o = new SimpleComposite("o"),
+          p = new SimpleComposite("p"),
+          new SimpleComposite("q",
+            u = new SimpleComposite("u"),
+            v = new SimpleComposite("v"),
+            w = new SimpleComposite("w"),
+            x = new SimpleComposite("x"),
+            y = new SimpleComposite("y")))));
+  }
+
+  @Test
+  public void same() {
+    SimpleComposite root = new SimpleComposite("root");
+    assertEquals(asList(root), Composites.allBetween(root, root));
+
+    assertEquals(asList(g), Composites.allBetween(g, g));
+  }
+
+  @Test(expected=IllegalArgumentException.class)
+  public void nonExisting() {
```

Comment:
may be some more tests for illegal cases will be useful, like allBetween(f, e) or allBetween(b, a)